tools/chplcheck/src/rules.py

```diff
@@ -323,3 +323,111 @@ def variables(node):
 
         for unused in indices.keys() - uses:
             yield indices[unused]
+
+    @driver.advanced_rule
+    def IncorrectIndentation(context, root):
+        """
+        Warn for inconsistent or missing indentation
+        """
+
+        # First, recurse and find warnings in children.
+        for child in root:
+            yield from IncorrectIndentation(context, child)
+
+        def contains_statements(node):
+            """
+            Returns true for allow-listed AST nodes that contain
+            just a list of statements.
+            """
+            return isinstance(node, (Record, Class, Module, Block))
```

Comment:
This would also include `Interface`, `Union`, `Enum`, and `Cobegin`. Also, why only `Block` and not `SimpleBlockLike`, you are excluding stuff like `On` and `Begin`